src/kernel/chainparams.cpp

```diff
@@ -542,3 +543,33 @@ std::unique_ptr<const CChainParams> CChainParams::TestNet()
 {
     return std::make_unique<const CTestNetParams>();
 }
+
+std::vector<int> CChainParams::GetAvailableSnapshotHeights() const {
+    std::vector<int> heights;
+    heights.reserve(m_assumeutxo_data.size());
+
+    for (const auto& data : m_assumeutxo_data) {
+        heights.emplace_back(data.height);
+    }
+    return heights;
+}
+
+std::optional<std::string> GetNetworkForMagic(MessageStartChars& message) {
+    const auto mainnet_msg = CChainParams::Main()->MessageStart();
+    const auto testnet_msg = CChainParams::TestNet()->MessageStart();
+    auto regtest_opts = CChainParams::RegTestOptions{};
+    const auto regtest_msg = CChainParams::RegTest(regtest_opts)->MessageStart();
+    auto signet_opts = CChainParams::SigNetOptions{};
+    const auto signet_msg = CChainParams::SigNet(signet_opts)->MessageStart();
+
+    if (std::equal(message.begin(), message.end(), mainnet_msg.data())) {
+        return "Mainnet";
```

Comment:
I'm surprised we never had to match the magic bytes to a chaintype. Instead of defining new strings here though, could you use `ChainTypeToString(ChainType::MAIN)`?

Reply:
Me too :) Yes, that makes sense. I opted for the function to return the `ChainType` and then I use `ChainTypeToString` on the result.